astropy/stats/bayesian_blocks.py

```diff
@@ -439,11 +466,16 @@ class Events(FitnessFunc):
         If ``ncp_prior`` is specified, ``gamma`` and ``p0`` is ignored.
     """
 
-    def fitness(self, N_k, T_k):
+    def fitness(self, N_k: NDArray[float], T_k: NDArray[float]) -> NDArray[float]:
         # eq. 19 from Scargle 2013
         return N_k * (np.log(N_k / T_k))
 
-    def validate_input(self, t, x, sigma):
+    def validate_input(
+        self,
+        t: ArrayLike,
+        x: ArrayLike | None,
+        sigma: float | ArrayLike | None,
+    ) -> tuple[ArrayLike, float | ArrayLike, float | ArrayLike]:
```

Comment:
I think this needs to be updated.

Reply:
Fixed in [51ffa26](https://github.com/astropy/astropy/pull/16562/commits/51ffa268c9eb79ebce85b9b56fca7ff92d4db8e2)